### PR TITLE
feat: Adds `Tag` component

### PR DESCRIPTION
**Pull request by hellofanny**
## What's the purpose of this pull request?

- Creates Tag component on @faststore/components 
- Adds Tag styles on @faststore/ui
- Adds Tag documentation page on docusaurus

## How it works?

It's composed by `Badge` + `Icon`

## How to test it?

run yarn start in app/docs folder and checks the `Tag` component or test it using the doc [preview link.](https://faststore-git-feat-tag-component-fs-605-faststore.vercel.app/reference/ui/molecules/Tag)

### CODE REVIEW

File: apps/docs/docs/reference/ui/molecules/Tag.mdx [hunk 1]
```diff
@@ -0,0 +1,81 @@
+---
+description: Interactive badges.
+keywords:
+  - UI library
+  - ecommerce
+  - react
+sidebar_custom_props:
+  image: https://vtexhelp.vtexassets.com/assets/docs/src/Badge___bcedffeb307527c1606be224a7fb5ac1.png
```
Comment (hellofanny) [inline apps/docs/docs/reference/ui/molecules/Tag.mdx:8]: todo: Update this image later.

File: apps/docs/docs/reference/ui/molecules/Tag.mdx [hunk 2]
```diff
@@ -0,0 +1,81 @@
+---
+description: Interactive badges.
```
Comment (hellofanny) [inline apps/docs/docs/reference/ui/molecules/Tag.mdx:2]: todo: Improve description

File: packages/components/src/molecules/Tag/Tag.tsx [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+
+const Tag = forwardRef<HTMLDivElement, TagProps>(function Tag(
+  { testId = 'fs-tag', label, icon, iconButtonLabel, onClose, ...otherProps },
+  ref
+) {
+  return (
+    <Badge ref={ref} data-fs-tag data-testid={testId} {...otherProps}>
+      <IconButton
+        data-fs-tag-icon-button
+        icon={icon ? icon : <X />}
```
Comment (hellofanny) [inline packages/components/src/molecules/Tag/Tag.tsx:38]: Left default icon button as cross icon.

File: packages/ui/src/components/molecules/Tag/styles.scss [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+      background-color: var(--fs-color-neutral-7);
+      filter: opacity(0.10);
+    }
+
+    &:hover, &:active, &:focus, &:focus-visible {
+      color: var(--fs-tag-text-color);
+      background-color: transparent;
+    }
+
+    svg {
```
Comment (hellofanny) [inline packages/ui/src/components/molecules/Tag/styles.scss:51]: Should this be tokens? 🤔

File: packages/ui/src/components/molecules/Tag/styles.scss [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+
+  [data-fs-badge-wrapper] {
+    position: relative;
+    display: flex;
+    align-items: center;
+    padding-right: var(--fs-spacing-6);
+    line-height: initial;
+  }
+
+  [data-fs-tag-icon-button] {
```
Comment (renatamottam) [inline packages/ui/src/components/molecules/Tag/styles.scss:25]: In order to button's tap size works properly, we need to add a `overflow: visible` here.

File: packages/ui/src/components/molecules/Tag/styles.scss [hunk 6]
```diff
@@ -0,0 +1,99 @@
...
+    position: absolute;
+    z-index: 1;
+    right: calc(-1 * var(--fs-spacing-1));
+    display: flex;
+    align-items: center;
+    justify-content: center;
+    width: var(--fs-control-tap-size);
+    height: var(--fs-control-tap-size);
```
Comment (renatamottam) [inline packages/ui/src/components/molecules/Tag/styles.scss:33]: This `--fs-control-tap-size` is rendering `54px`, shouldn't it be `48px`?

**Review by renatamottam — APPROVED** [review]
Loved the improvement proposals! Nice one, fan :)

File: packages/ui/src/components/molecules/Tag/styles.scss (hunk 6 above)
Comment (hellofanny) [reply packages/ui/src/components/molecules/Tag/styles.scss:32]: I noticed that this is a general problem, trying to fix it :)

File: apps/docs/docs/reference/ui/molecules/Tag.mdx (hunk 1 above)
Comment (hellofanny) [reply apps/docs/docs/reference/ui/molecules/Tag.mdx:8]: @carolinamenezes can u please update this image later too? thanks

The [component](https://faststore-git-feat-tag-component-fs-605-faststore.vercel.app/reference/ui/molecules/Tag) looks like this:
Badge with a close button

image

File: packages/ui/src/components/molecules/Tag/styles.scss (hunk 5 above)
Comment (hellofanny) [reply packages/ui/src/components/molecules/Tag/styles.scss:25]: hmm. I didn't get this one 👀 
do we still need it here?

File: packages/components/src/molecules/IconButton/IconButton.tsx [hunk 7]
```diff
@@ -37,7 +37,7 @@ const IconButton = forwardRef<HTMLButtonElement, IconButtonProps>(
         data-fs-button
         data-fs-icon-button
         testId={testId}
-        variant={variant}
+        variant={variant ? variant : 'tertiary'}
```
Comment (lucasfp13) [inline packages/components/src/molecules/IconButton/IconButton.tsx:40]:
```suggestion
        variant={variant ?? 'tertiary'}
```

**Review by lucasfp13 — APPROVED** [review]

**Pull request merged**